src/plone/restapi/tests/test_blocks_searchable_text.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+from plone.app.testing import setRoles
+from plone.app.testing import SITE_OWNER_NAME
+from plone.app.testing import SITE_OWNER_PASSWORD
+from plone.app.testing import TEST_USER_ID
+from plone.dexterity.interfaces import IDexterityFTI
+from plone.dexterity.utils import createContentInContainer
+from plone.restapi.testing import PLONE_RESTAPI_DX_FUNCTIONAL_TESTING
+from plone.restapi.testing import RelativeSession
+from zope.component import queryUtility
+
+import transaction
+import unittest
+
+
+class TestSearchTextInBlocks(unittest.TestCase):
+
+    layer = PLONE_RESTAPI_DX_FUNCTIONAL_TESTING
+
+    def setUp(self):
+        self.app = self.layer["app"]
+        self.portal = self.layer["portal"]
+        self.portal_url = self.portal.absolute_url()
+        setRoles(self.portal, TEST_USER_ID, ["Manager"])
+
+        self.api_session = RelativeSession(self.portal_url)
+        self.api_session.headers.update({"Accept": "application/json"})
+        self.api_session.auth = (SITE_OWNER_NAME, SITE_OWNER_PASSWORD)
+
+        fti = queryUtility(IDexterityFTI, name="Document")
+        behavior_list = [a for a in fti.behaviors]
+        behavior_list.append("volto.blocks")
+        fti.behaviors = tuple(behavior_list)
+
+        self.doc = createContentInContainer(
+            self.portal, u"Document", id=u"doc", title=u"A document"
+        )
+        transaction.commit()
```

Comment:
I think you are calling commit to have the catalog queue processed.
It is better then to call the `processQueue` function.

Reply:
When we use a functional test, does that make much of a difference?